### ISSUE TITLE
Support for Bottlerocket as a custom AMI in managed node groups

### ISSUE THREAD

**Issue by mikestef9** [weaveworks/eksctl#3033]
**Why do you want this feature?**
Use Bottlerocket, the AWS container optimized system, as a custom AMI with managed node groups

**What feature/behavior/change do you want?**
I would like to create a Bottlerocket node group through managed node groups using eksctl. Currently, this does not work because eksctl injects MIME metadata into the user data. MIME is not required when using a custom AMI with managed node groups. ekctl should just pass the `overrideBootstrapCommand` field value as is to managed node groups API

I would expect the below config to work with this update

```
managedNodeGroups:
  - name: x86-bottlerocket-mng
    ami: ami-0acfc3fffaaa4e015
    instanceType: m5.large
    minSize: 1
    desiredCapacity: 2
    maxSize: 3
    privateNetworking: true
    disableIMDSv1: true
    volumeSize: 50
    labels: { role: x86-bottlerocket-mng }
    iam:
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
    overrideBootstrapCommand: |
      [settings.kubernetes]
      api-server = "https://redacted.gr7.us-east-1.eks.amazonaws.com"
      cluster-certificate = "redacted"
      cluster-name = "testing-cluster"
```

**Comment by michaelbeaumont** [comment]
@mikestef9 At the moment, we use MIME data/cloud-init not only to bootstrap nodes on unmanaged/managed nodegroups but also to e.g. install the SSM agent.

See: https://github.com/weaveworks/eksctl/blob/501b67a89b84fc21c880f9f9a467fa8e80d9ba76/pkg/cfn/builder/managed_launch_template.go#L143-L148 and https://github.com/weaveworks/eksctl/blob/501b67a89b84fc21c880f9f9a467fa8e80d9ba76/pkg/cfn/builder/managed_launch_template.go#L168-L185

If we don't have MIME multi-part for custom AMIs + managed nodegroups in general, we can no longer support this use case, as far as I can tell.

It seems like how the user data is handled (and whether MIME is supported) depends on the AMI itself.
The only option therefore is that, when `ami: ` is set, _only_ allow `overrideBootstrapCommand` to be set and disallow any options that depend on adding scripts to the user data.

Does that make sense?

**Comment by mikestef9** [comment]
Right, if `ami:` is set, that indicates use of a custom AMI with managed node groups. And when using a custom AMI, managed node groups does not validate or care if user data is in MIME format. And the bootstrap is fully in control of the user.

**Comment by cPu1** [comment]
>It seems like how the user data is handled (and whether MIME is supported) depends on the AMI itself.
The only option therefore is that, when ami: is set, only allow overrideBootstrapCommand to be set and disallow any options that depend on adding scripts to the user data.

That's true, which is why we disallow features in MNG that add scripts to user data (like `maxPodsPerNode`). We missed this when adding the SSM agent feature, but it should not be allowed when a custom AMI is provided.

**Comment by mikestef9** [comment]
Confirmed this is resolved with eksctl 0.37.0. Thanks!
